### DISCUSSION TITLE
Red coverage reports for covered code

### DISCUSSION THREAD

**Discussion by kieranm** [vitest-dev/vscode#457] [Q&A]
Hi all,

Does anyone know why the coverage report would highlight code in red here even though we can see it's being covered?

<img width="998" alt="Screenshot 2024-08-20 at 10 38 39" src="https://github.com/user-attachments/assets/7f95909a-d54d-4bbc-b9ee-e644fdc9ff5d">

The breakpoints are all being hit and we can see the `1x` next to each line indicating that it was captured during the test so I'm a little confused here.

Thanks!

**Comment by kieranm** [comment]
If I run certain tests in isolation then I can get this block of code to show up in blue:
<img width="910" alt="Screenshot 2024-08-20 at 10 45 21" src="https://github.com/user-attachments/assets/3b156d25-e150-42cb-bdeb-a51b522a0b0e">

but when I run a bunch of tests together in the UI, it's like the coverage reports overwrite each other instead of getting combined with each other.

Is there a setting I might be missing here? The output of the `vitest` command looks correct in terms of the coverage it outputs.

**Reply by sheremet-va** [reply]
Each test run has a unique coverage report, they are not merged

**Reply by kieranm** [reply]
So it's not possible to see what code has not been tested across any of the tests?

**Reply by sheremet-va** [reply]
If you want to see the same coverage you see in your terminal, you need to run all tests

**Reply by kieranm** [reply]
I am running all the tests using this button and the output isn't the same as the terminal

<img width="397" alt="Screenshot 2024-08-20 at 10 53 03" src="https://github.com/user-attachments/assets/987f7e2c-121a-441f-b724-29301b71ec1d">

**Reply by kieranm** [reply]
Also you can see the `1x` beside each line in my first screenshot so it seems to be being captured correctly by coverage? I just don't know why the colour is wrong

**Reply by sheremet-va** [reply]
I cannot help you further. The extension passes the same json generated file to display results using vscode's default coverage handler: https://code.visualstudio.com/docs/editor/testing#_test-coverage

You can create an issue with a reproduction
